src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUtil.java

```diff
@@ -336,4 +337,20 @@ public static void removeRedundantProperties(final Dictionary<String, Object> pr
             }
         }
     }
+
+    public static String getPid(final ConfigurationEvent event) {
+        final String id;
+        final String pid;
+        if (event.getFactoryPid() == null ) {
```

Comment:
I think the code would be clearer if you would check if you need to after the pid ( branch on line 347 ) and then have a fallback to `id = event.getPid();`.